### PR TITLE
Adding CRDB into GHA for the new store

### LINKED ISSUE #16020: Adding CRDB into GHA for the new store
The CockroachDB database is slower than PostgreSQL, and while it is using almost the same Java JPA code for the new store, it has its own different behavior as everything is SERIALIZABLE, and rollbacks might occur more frequently than on PostgreSQL with REPEATABLE_READ.

This issue will enable running the same tests as for PostgreSQL on branches and nightlies to give enough feedback to the storage team when CockroachDB breaks. 

For the sake of saving some time, it will not run on pull requests. 

If there are CockroachDB related changes in a PR, the reviewer and the contributor should have a look in the forked repository (knowing that this has not been rebased, still it seems to be sufficient).

### PR DESCRIPTION
**Pull request by ahus1**
The CockroachDB database is slower than PostgreSQL, therefore it will only run branches and nightly builds.

Closes #16020

### CODE REVIEW

**Comment by ahus1** [comment]
@sguilhen - here's the GHA for CRDB wich will run only on the nightly and on branches ready for review.

There's a run in my repository where it ran on a branch, with CRDB enabled: https://github.com/ahus1/keycloak/actions/runs/3705121796

image

As I'm writing this the model tests again, they where unstable for a different reason. In this PR, they succeeded.

File: testsuite/model/src/test/java/org/keycloak/testsuite/model/parameters/JpaMapStorageCockroachdb.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
```
Comment (sguilhen) [inline testsuite/model/src/test/java/org/keycloak/testsuite/model/parameters/JpaMapStorageCockroachdb.java:2]: nit: year

File: testsuite/integration-arquillian/HOW-TO-RUN.md [hunk 2]
```diff
@@ -859,7 +859,7 @@ podman run --name postgres -p 5432:5432 -e POSTGRES_PASSWORD=pass -e POSTGRES_US
 To run the tests without spawning the container for you, execute tests with the following command:
 ```shell
 mvn clean install -f testsuite/integration-arquillian/tests/base \
-  -Pmap-storage,map-storage-jpa \
+  -Pmap-storage,map-storage-jpa-postgres \
   -Dpostgres.start-container=false \
   -Dkeycloak.map.storage.connectionsJpa.url=<jdbc_url> \
   -Dkeycloak.map.storage.connectionsJpa.user=<user> \
```
Comment (sguilhen) [inline testsuite/integration-arquillian/HOW-TO-RUN.md:865]: Do you think it is worth to briefly mention `map-storage-jpa-cockroach` also as a valid profile that can be used to run the tests with CRDB?

**Comment by sguilhen** [comment]
> @sguilhen - here's the GHA for CRDB wich will run only on the nightly and on branches ready for review.
> 

Thanks @ahus1 , I really like the changes being proposed here! I only added a couple of comments, really minor things.

**Comment by stianst** [comment]
@ahus1 How much slower is it?

File: testsuite/integration-arquillian/HOW-TO-RUN.md (hunk 2 above)
Comment (ahus1) [reply testsuite/integration-arquillian/HOW-TO-RUN.md:870]: Updated in latest push.

**Comment by ahus1** [comment]
> How much slower is it?

@stianst - looking at https://github.com/ahus1/keycloak/actions/runs/3705121796, the run time were: 

CHM: 15 min
HotRod: 18 min
JPA PostgreSQL: 41 min
JPA CockroachDB: 48 min

The executions time vary a bit, and I've seen JPA PostgreSQL completing in 35 min. before. I didn't do lots of runs, still I think it is 20-30% slower. 

Best,
Alexander

**Comment by sguilhen** [comment]
JPA PostgreSQL usually finishes in around 30-35 mins. JPA CockroachDB takes over 45 mins, which is 40 to 50% slower. If we change the set of tests that are run with JPA in the future to include more tests, the difference can get bigger (entire testsuite takes twice as much to run on CRDB compared to PostgreSQL).

**Comment by ahus1** [comment]
@sguilhen - tests have passed. Ready for the next review round.

**Review by sguilhen — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -156,14 +156,31 @@ jobs:
...
 
+  databases-new-store:
+    name: Databases New Store
+    runs-on: ubuntu-latest
+    steps:
+      - id: matrix-db
+        # language=bash
+        run: |
+          # For PRs, run only PostgreSQL, on branches and nightly runs run all databases
+          if [ "${GITHUB_EVENT_NAME}" == "pull_request" ]; then
```
Comment (stianst) [inline .github/workflows/ci.yml:167]: Nitpick, but I prefer to use ${{ github.event_name }} as that allows you to see the value when debugging, while using "${GITHUB_EVENT_NAME}" you don't see the value AFAIK

Comment (ahus1) [reply .github/workflows/ci.yml]: Changes, and also rebased due to conflicts that occurred when other changes were merged into main.

**Review by stianst — APPROVED** [review]

**Pull request merged**